hostplumber/controllers/hostconfig_controller.go

```diff
@@ -83,12 +84,103 @@ func (r *HostNetworkTemplateReconciler) Reconcile(req ctrl.Request) (ctrl.Result
 		log.Info("SriovConfig is empty")
 	}
 
+	// Everything that is traditonally done under "ifconfig <ifname>" handled here
+	// Alternatively newer "ip addr" and "ip link" - see https://www.redhat.com/sysadmin/ifconfig-vs-ip
+	// MTUs, IPs, routes, link up/down, etc...
+	ifConfigList := hostConfigReq.Spec.InterfaceConfig
+	if len(ifConfigList) > 0 {
+		if err := applyInterfaceConfig(ifConfigList); err != nil {
+			log.Error(err, "Failed to apply interfaceConfig")
+			return ctrl.Result{}, err
+		}
+	} else {
+		log.Info("interfaceConfig is empty")
+	}
+
 	hni := hoststate.New(r.NodeName, r.Namespace, r.Client)
 	hni.DiscoverHostState()
 
 	return ctrl.Result{}, nil
 }
 
+func applyInterfaceConfig(ifConfigList []plumberv1.InterfaceConfig) error {
+	for _, ifConfig := range ifConfigList {
+		var ifName string
+
+		// TODO: nil check needed? This does NOT have an omitempty json tag in _types.go
```

Comment:
If it is a pointer, and it is, then yes 😄 You shouldn't rely on the JSON parser to handle this case for you; if someone adds the omitempty tag or uses this function in another way, they will probably not expect this behaviour.